server/controllers/report.js

```diff
@@ -607,16 +607,34 @@ function expenseReport (params) {
 
 function allTrans (params){
   var source = {
+    '0' : 0,
     '1' : 'posting_journal',
     '2' : 'general_ledger'
   };
   var def = q.defer();
   params = JSON.parse(params);
   var requette;
-  var suite_account = (params.account_id && params.account_id !== 0)? ' AND `t`.`account_id`=' + sanitize.escape(params.account_id) : '';
+  var suite_account = (params.account_id && params.account_id > 0)? ' AND `t`.`account_id`=' + sanitize.escape(params.account_id) : '';
   var suite_dates = (params.datef && params.datet)? ' AND `t`.`trans_date`>= ' + sanitize.escape(params.datef) + ' AND `t`.`trans_date`<= ' + sanitize.escape(params.datet) : '';
 
-  if (!params.source || params.source === 0) {
+  if (params.source && params.source > 0) {
+    var sub_chaine = [
+      '`uuid`, ', '`inv_po_id`, ',
+      '`trans_date`, ', '`debit_equiv`, ',
+      '`credit_equiv`, ', '`account_id`, ',
+      '`deb_cred_uuid`, ', '`currency_id`, ', '`doc_num`, ',
+      '`trans_id`, ', '`description`, ', '`comment` '
+    ].join(source[params.source] + '.');
+    sub_chaine = source[params.source] + '.' + sub_chaine;
+
+    requette =
+      'SELECT `t`.`uuid`, `t`.`trans_id`, `t`.`trans_date`, `ac`.`account_number`, `t`.`debit_equiv` AS `debit`,  ' +
```

Comment:
Use a semantic name, instead of "t".  You can also remove the "`" tick marks, they do not do anything.